openedx/core/djangoapps/programs/tests/test_tasks.py

```diff
@@ -0,0 +1,244 @@
+        )
+        payload = [
+            {'course_id': 'test-course-1', 'mode': 'verified'},
+            {'course_id': 'test-course-2', 'mode': 'prof-ed'},
+        ]
+        result = tasks.get_completed_programs(test_client, payload)
+        self.assertEqual(httpretty.last_request().body, json.dumps(payload))
+        self.assertEqual(result, [1, 2, 3])
+
+
+class GetAwardedCertificateProgramsTestCase(TestCase):
+    """
+    Test the get_awarded_certificate_programs function
+    """
+
+    def make_credential_result(self, **kwargs):
+        """
+        Helper to make dummy results from the credentials API
+        """
+        result = {
+            'id': 1,
+            'username': 'dummy-username',
+            'credential': {
+                'credential_id': None,
+                'program_id': None,
+            },
+            'status': 'dummy-status',
+            'uuid': 'dummy-uuid',
+            'certificate_url': 'http://credentials.edx.org/credentials/dummy-uuid/'
+        }
+        result.update(**kwargs)
+        return result
+
+    @mock.patch('openedx.core.djangoapps.programs.tasks.get_user_credentials')
+    def test_get_awarded_certificate_programs(self, mock_get_user_credentials):
+        """
+        Ensure the API is called and results handled correctly.
+        """
+        student = UserFactory(username='test-username')
+        mock_get_user_credentials.return_value = [
+            self.make_credential_result(status='awarded', credential={'program_id': 1}),
+            self.make_credential_result(status='awarded', credential={'course_id': 2}),
+            self.make_credential_result(status='revoked', credential={'program_id': 3}),
+        ]
+
+        result = tasks.get_awarded_certificate_programs(student)
+        self.assertEqual(mock_get_user_credentials.call_args[0], (student, ))
+        self.assertEqual(result, [1])
+
+
+class AwardProgramCertificateTestCase(TestCase):
+    """
+    Test the award_program_certificate function
+    """
+
+    @httpretty.activate
+    def test_award_program_certificate(self):
+        """
+        Ensure the correct API call gets made
+        """
+        student = UserFactory(username='test-username')
+        test_client = EdxRestApiClient('http://test-server', jwt='test-token')
+        httpretty.register_uri(
+            httpretty.POST,
+            'http://test-server/user_credentials/',
+        )
+        tasks.award_program_certificate(test_client, student, 123)
+        self.assertEqual(httpretty.last_request().body, json.dumps({'program_id': 123, 'username': student.username}))
+
+
+@ddt.ddt
+@mock.patch('openedx.core.djangoapps.programs.tasks.award_program_certificate')
+@mock.patch('openedx.core.djangoapps.programs.tasks.get_awarded_certificate_programs')
+@mock.patch('openedx.core.djangoapps.programs.tasks.get_completed_programs')
+@mock.patch('openedx.core.djangoapps.programs.tasks.get_completed_courses')
```

Comment:
it may be for lack of imagination on my part but i found httpretty kind of difficult to work with especially with multiple API calls.  That's part of why I have things structured this way.  I was running into a lot of trouble debugging httpretty when it didn't seem to be catching the uris i was registering and this was taking a lot of time.

I feel like part of the value of using the slumber client is that we can trust it to translate method/function calls into HTTP interactions, which is another reason that I think httpretty, while very useful in various situations, isn't absolutely essential for validating correct usage of the APIs.  But I'm open to changing if your argument is that coverage/confidence will improve if we switch to doing that.